metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -229,6 +230,11 @@ final class BuildTargets() {
     }
   }
 
+  def jvmRunEnvironment(
+      targetId: BuildTargetIdentifier
+  ): Option[JvmEnvironmentItem] =
+    data.fromOptions(_.jvmRunEnvironments.get(targetId))
```

Comment:
I changed the behaviour to save the jvm data, since it will only be changed on reconnecting.